pctasks/core/pctasks/core/storage/blob.py

```diff
@@ -539,15 +539,25 @@ def upload_file(
         input_path: str,
         target_path: str,
         overwrite: bool = True,
+        **kwargs: Any,
     ) -> None:
+        """
+        Upload a file to blob storage.
+
+        Parameters
+        ----------
+        **kwargs:
+            Additional keyword arguments are provided to
+            :class:`azure.storage.blob.BlobClient.upload_blob`.
+        """
         with self._get_client() as client:
             with client.container.get_blob_client(
                 self._add_prefix(target_path)
             ) as blob:
 
                 def _upload() -> None:
                     with open(input_path, "rb") as f:
-                        blob.upload_blob(f, overwrite=overwrite)
+                        blob.upload_blob(f, overwrite=overwrite, **kwargs)
```

Comment:
This is useful for setting the content type in Blob Storage. We do that for the JPEG thumbnails we generate and upload.

Reply:
Perhaps this should be more explicit rather than kwargs. E.g. have a `content_type: Optional[str] = None` as an argument. That would make it more clear to use in other cases. Otherwise the user has to know this is the mechanism for content type settings and deal with ContentSettings, and know to set them in kwargs, whereas having it directly in the signature would be easier to use.

Reply:
Very fair. My only hesitation is that `BlobClient.upload_blob` has a lot of kwargs, and it's not obvious where you stop. But given a couple of years of experience and this is the only kwarg we've seen a need for, maybe that's where we drop the line.

Reply:
Fixed in 1522a0d8e7900c197f5fb8dd615609dfe0f4d1d0